### DISCUSSION TITLE
Where does MeshCommander's node installed readme.txt come from?

### DISCUSSION THREAD

**Discussion by crass** [Ylianst/MeshCentral#4776] [General]
I've installed MeshCommander using node as suggested in the its [readme](https://github.com/Ylianst/MeshCommander/blob/master/readme.md). It indicates that its version 0.9.5. Looking at the files that node downloaded I see a readme.txt file at `node_modules/meshcommander/readme.txt`. This is **not** the readme from the git repo, nor does it seem to ever have been. There doesn't seem to be anything otherwise strange about the readme, but it makes me wonder what else it being included in the node install that isn't in the git repo. @Ylianst what's going on here?

Here's a section it has that I've not found elsewhere:
```
Tutorials
---------

Introduction to MeshCommander.

[![MeshCommander - Introduction](https://img.youtube.com/vi/k7xVkZSVY0E/mqdefault.jpg)](https://www.youtube.com/watch?v=k7xVkZSVY0E)


Using MeshCommander's hardware KVM viewer.

[![MeshCommander - Usages](https://img.youtube.com/vi/5EKw1g0L3IE/mqdefault.jpg)](http://www.youtube.com/watch?v=5EKw1g0L3IE)


Setting up Intel&reg; AMT with TLS.

[![MeshCommander - Intel AMT CIRA](https://img.youtube.com/vi/2WYanQ11gYE/mqdefault.jpg)](http://www.youtube.com/watch?v=2WYanQ11gYE)
```

**Comment by jsastriawan** [comment]
Meshcommander has several versions. It can be nodejs web app version, nodewebkit or a compact version which can be uploaded to AMT file storage so that it can be hosted by AMT firmware itself. Each version is coming through compilation process. This process is manual using Website Compiler. So, I assume Ylian has some offline packaging mechanism not hosted by this repo. Not sure if Ylian has YouTube tutorial on Website Compiler or not but this is how I used to work on Meshcommander code when I add new features.

**Reply by crass** [reply]
Ok, that's all fine and good and about what I expected, but it doesn't really answer the question. Why is there extra data that is not from the source repository? What if instead of the readme being different it was extra code which sent usage statistics back to Intel without the user being asked? I would consider that a breach of trust. This is an open source project and people expect that releases of that project were built from the source repository without anything extra. Ideally the user should be able to follow build instructions to reproduce a release to verify that.

Now I don't think anything malicious is going on and I am impressed with the project. However, this indicates some weirdness in the release build process, perhaps it needs to be cleaned up. Everything installed by npm in the `node_modules/meshcommander/` directory should originate from the git repo. The readme for MeshCentral **is** the one located in the git repo.
